platform-sdk/swirlds-state-impl/src/main/java/com/swirlds/state/merkle/disk/OnDiskReadableSingletonState.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2020-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.state.merkle.disk;
+
+import com.hedera.pbj.runtime.Codec;
+import com.swirlds.state.spi.ReadableSingletonStateBase;
+import com.swirlds.virtualmap.VirtualMap;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+import static com.swirlds.state.merkle.StateUtils.computeLabel;
+import static com.swirlds.state.merkle.StateUtils.getVirtualMapKey;
+import static com.swirlds.state.merkle.logging.StateLogger.logSingletonRead;
+import static java.util.Objects.requireNonNull;
+
+public class OnDiskReadableSingletonState<T> extends ReadableSingletonStateBase<T> {
+
+    @NonNull
+    private final VirtualMap virtualMap;
+
+    @NonNull
+    private final Codec<T> valueCodec;
+
+    public OnDiskReadableSingletonState(
+            @NonNull final String serviceName,
+            @NonNull final String stateKey,
+            @NonNull final Codec<T> valueCodec,
+            @NonNull final VirtualMap virtualMap) {
+        super(serviceName, stateKey);
+
+        this.virtualMap = requireNonNull(virtualMap);
+        this.valueCodec = requireNonNull(valueCodec);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected T readFromDataSource() {
+        final var value = virtualMap.get(getVirtualMapKey(serviceName, stateKey), valueCodec);
```

Comment:
VirtualMap key for singletons is a constant, no need to calculate it every time

Reply:
I addressed it, but it is cached anyway in `StateUtils`

Reply:
It is cached in `StateUtils` anyway, so if I start updating it, it will bloat the code in the multiple classes (where a similar update can be introduced), which is redundant, in my opinion. Wdyt?